cmd/manager/main.go

```diff
@@ -118,6 +124,9 @@ func main() {
 		exit(err)
 	}
 
+	// Generate metrics and publish it at an endpoint.
```

Comment:
Should this be async? Though I suppose we care that we start recording before we start handling events so we don't miss any? Have we considered what all this initialization will do (in terms of time cost) if we don't at least serve the health check first?

Reply:
The `RecordMetrics()` function uses the endpoint defined by SDK. Hence the health check does happen here before the metrics are served. The scrapping of data by Prometheus is synchronous. But yes, as you said we can decouple recording metrics logic from this function and start collecting metrics initially. Creation of service and service monitor could happen later.